Build for phageParser with Make

kinow opened the pull request:
Hi, 

This pull request introduces a simple Makefile, based on [Khmer](https://github.com/ged-lab/khmer)'s Makefile - tis convenient to be able to use another good quality tool written in Python as reference :-)

I tried removing what was specific to Khmer, leaving simple targets like help, clean, test and the static analysis code tools (pep8 and pep257).

If this pull request gets merged, I'd like to either send another pull request, or talk during the mozilla sprint about moving Python files to packages, creating classes and making it possible to call the filter/parser/etc functions without having to change the code, by passing parameters (e.g. filterByExpect.py).

Thank you!

BillMills commented:
hey @kinow - those ideas all sound good to me! I will dig into this PR on the weekend and comment then.

kinow commented:
Great, thanks @BillMills !

BillMills commented on Makefile:
```diff
@@ -0,0 +1,71 @@
+# The MIT License (MIT)
```
no need to repeat the license here

BillMills commented on NOTICE:
```diff
@@ -0,0 +1,3 @@
+Parts of the Makefile were adapted from the Khmer project one.
```
move this into the top of the makefile, instead of the license text

BillMills commented on filter/__init__.py:
```diff
@@ -0,0 +1,22 @@
+# The MIT License (MIT)
```
again, no need to repeat licenses

BillMills commented:
Hey @kinow, this looks great at first pass (tests, hooray!) - most important thing to add would be an explanation in README on how to use all this fancy new machinery, and a short comment on what it's doing, for people who have never used `make` before. Thanks for this!

kinow commented:
Hi @BillMills! Thanks for the feedback!

>(tests, hooray!)

The makefile has the basic structure for running tests written with nose in Python. I plan to later write basic tests for the filterByExpect.py script (I'm focusing in the first step in the tutorial for now). That work is probably going to be tracked in the issue #18

>most important thing to add would be an explanation in README on how to use all this fancy new machinery, and a short comment on what it's doing, for people who have never used make before. Thanks for this!

I've added a basic paragraph in the README.md, saying that phageParser uses Make, and a link to the manual. Probably a Wiki page (e.g. Building phageParser) could contain more detailed instructions. I like the way that the README.md file explains what is the project and how to start using it. Putting a lot of links about Make, and subsections to understand the targets, dependencies, etc; I think that would distract new users. What do you think?

I've also added a new target `filter_by_expect` and changed the filterByExpect.py script so that it supports parameters and uses Python logging. This way it is easier to run the script with several different input files or threshold, and compare the output.

mbonsma commented:
Looks cool! I'm one of the `make` newbies @BillMills is referring to. If we end up with more detailed instructions, it would be great if they had a blurb about using `make` with Windows. So far I haven't got it to work, but I'm pretty sure it's an issue with syntax in the Windows command prompt and not anything else (I'm using GnuWin32 which is supposed to allow `make` to be used from the command line).

kinow commented:
Hi @mbonsma I'll write a short Wiki page for that. And though my Windows-fu is quite rusty, maybe I can help debugging the problems you have. Do you usually lurk in any IRC channel, or mailing list? Or maybe you can file an issue to the repo with the problems you have with Make, and I'll try to figure a workaround. Thanks!

mbonsma commented:
Thanks @kinow! I made an issue ( #34 ).
